### PR TITLE
[desktop] Handle encryption errors for config

### LINKED ISSUE #3111: Invalid mac on desktop
A lot of reports still coming with invalid mac.

We should reset all stored data if this happens as key/data might get out of sync.

```
Tutanota version: 3.83.3
User agent:
Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) tutanota-desktop/3.83.3 Chrome/87.0.4280.141 Electron/11.4.4 Safari/537.36
CryptoError
Error message: invalid mac
Stacktrace:
CryptoError: invalid mac
at C:\Program Files\Tutanota Desktop\resources\app.asar\desktop\DesktopMain.js:1:415399
at Object.aes256Decrypt (C:\Program Files\Tutanota Desktop\resources\app.asar\desktop\DesktopMain.js:1:415739)
at Object.aesDecryptObject (C:\Program Files\Tutanota Desktop\resources\app.asar\desktop\DesktopMain.js:13:205269)
at Object._getEncryptedVar (C:\Program Files\Tutanota Desktop\resources\app.asar\desktop\DesktopMain.js:13:208726)
at async C:\Program Files\Tutanota Desktop\resources\app.asar\desktop\DesktopMain.js:13:246033
```

### PR DESCRIPTION
**Pull request by charlag**
fix #3111

### CODE REVIEW

File: src/desktop/DesktopCryptoFacade.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ export class DesktopCryptoFacade {
 		this._addEntropy()
 	}
 
-	aesEncryptObject(encryptionKey: Aes256Key, object: {}): string {
+	aesEncryptObject(encryptionKey: Aes256Key, object: number | string | boolean | $ReadOnlyArray<*> | {}): string {
```
Comment (johnbotris) [inline src/desktop/DesktopCryptoFacade.js:27]: isn't `{}` just `any` anyway?

File: src/desktop/config/DesktopConfig.js [hunk 2]
```diff
@@ -160,28 +179,12 @@ export class DesktopConfig {
...
-				const configKey: AllConfigKeysEnum = downcast(p)
-				if (configKey === 'any') continue;
-				if (this._onValueSetListeners[configKey]) {
-					for (const cb of this._onValueSetListeners[configKey]) {
-						cb(value[configKey])
-					}
-				}
-			}
-		} else if (this._onValueSetListeners[key]) {
+	_notifyChangeListeners(key: AllConfigKeysEnum, value: ?ConfigValue) {
```
Comment (johnbotris) [inline src/desktop/config/DesktopConfig.js:182]: i assume `"any"` was never being used?

**Review by johnbotris — APPROVED** [review]

File: src/desktop/DesktopCryptoFacade.js (hunk 1 above)
Comment (charlag) [reply src/desktop/DesktopCryptoFacade.js:27]: No it's not its object without requirements

File: src/desktop/config/DesktopConfig.js (hunk 2 above)
Comment (charlag) [reply src/desktop/config/DesktopConfig.js:182]: Nope

**Pull request merged**